lib/iris/fileformats/netcdf/_thread_safe_nc.py

```diff
@@ -35,22 +35,42 @@ class _ThreadSafeWrapper(ABC):
     the C-layer.
     """
 
+    # Note: this is only used to create a "contained" from passed args.
     CONTAINED_CLASS = NotImplemented
+    # Note: this defines how we identify/check that a contained is of the expected type
+    # (in a duck-type way).
+    _DUCKTYPE_CHECK_PROPERTIES: typing.List[str] = [NotImplemented]
 
     # Allows easy type checking, avoiding difficulties with isinstance and mocking.
     THREAD_SAFE_FLAG = True
 
+    @classmethod
+    def is_contained_type(cls, instance):
+        return all(
+            hasattr(instance, attr) for attr in cls._DUCKTYPE_CHECK_PROPERTIES
+        )
+
     @classmethod
     def _from_existing(cls, instance):
         """Pass an existing instance to __init__, where it is contained."""
-        assert isinstance(instance, cls.CONTAINED_CLASS)
+        assert cls.is_contained_type(instance)
         return cls(instance)
 
     def __init__(self, *args, **kwargs):
         """Contain an existing instance, or generate a new one from arguments."""
-        if isinstance(args[0], self.CONTAINED_CLASS):
+        if len(args) > 0 and self.is_contained_type(args[0]):
```

Comment:
```suggestion
        if len(args) == 1 and self.is_contained_type(args[0]):
```

This is a very deliberately designed use case. If this receives 2 or more arguments then the developer is going very much off-piste and not using as intended!

Reply:
My only concern was to avoid this causing an error with **_no_** arguments.  I think any of the actual intended contained classes will give an error then anyway, but at least it's then raised by the appropriate thing.

Reply:
Sorry got confused about this.  Yes, just one arg of an expected type is the thing.